Relax SC2182 for bash, when printf -v is used

Specifically for *bash*, `printf` has an extra option, `-v` which you can supply a variable name.  This named variable is populated with the result of the `printf` call.  This means you can write `printf -vvar '%s x %s' "$1" "$2"; echo "$var"`.  

Usually, the first arg to `printf` is a format string, so in POSIX shell, an equivalent might be `var="$(printf '%s x %s' "$1" "$1")"; echo "$var"`.

SC2182 fires for *bash* if the first arg is `-vvar`, because `-vvar` does not contain any format string tokens.  However, it is a valid argument to *bash*'s `printf`.

#### For bugs
- Rule Id (if any, e.g. SC1000): SC2182
- My shellcheck version (`shellcheck --version` or "online"): 0.9 (according to https://github.com/vscode-shellcheck/vscode-shellcheck/releases/tag/v0.29.2)
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
printf -vvar '%s x %s' "$1" "$2"
echo "$var"
```

#### Here's what shellcheck currently says:
[SC2182](https://www.shellcheck.net/wiki/SC2182) (error): This printf format string has no variables. Other arguments are ignored.


#### Here's what I wanted or expected to see:
no error

--- a/src/ShellCheck/Checks/Commands.hs
+++ b/src/ShellCheck/Checks/Commands.hs
@@ -678,10 +678,12 @@ prop_checkPrintfVar19 = verifyNot checkPrintfVar "printf '%(%s)T'"
 prop_checkPrintfVar20 = verifyNot checkPrintfVar "printf '%d %(%s)T' 42"
 prop_checkPrintfVar21 = verify checkPrintfVar "printf '%d %(%s)T'"
 prop_checkPrintfVar22 = verify checkPrintfVar "printf '%s\n%s' foo"
+prop_checkPrintfVar23 = verifyNot checkPrintfVar "printf -vTODAY '%(%Y)T'"
 
 checkPrintfVar = CommandCheck (Exactly "printf") (f . arguments) where
     f (doubledash:rest) | getLiteralString doubledash == Just "--" = f rest
     f (dashv:var:rest) | getLiteralString dashv == Just "-v" = f rest
+    f (dashvVar:rest) | Just ('-':'v':_:_) <- getLiteralString dashvVar = f rest
     f (format:params) = check format params
     f _ = return ()
 

